src/task.c

```diff
@@ -235,6 +235,52 @@ JL_DLLEXPORT void *jl_task_stack_buffer(jl_task_t *task, size_t *size, int *tid)
     return (void *)((char *)task->stkbuf + off);
 }
 
+JL_DLLEXPORT void jl_active_task_stack(jl_task_t *task, char **start, char **end)
+{
+    if (!task->started) {
+        *start = *end = 0;
+        return;
+    }
+
+    int16_t tid = task->tid;
+    jl_ptls_t ptls2 = (tid != -1) ? jl_all_tls_states[tid] : 0;
+
+    if (task->copy_stack && ptls2 && task == ptls2->current_task) {
+        *start  = (char*)ptls2->stackbase - ptls2->stacksize;
+        *end = (char*)ptls2->stackbase;
+    }
+    else if (task->stkbuf) {
+        *start  = (char*)task->stkbuf;
+#ifdef COPY_STACKS
+        // save_stack stores the stack of an inactive task in stkbuf, and the
+        // actual number of used bytes in copy_stack.
+        if (task->copy_stack > 1)
+            *end = (char*)task->stkbuf + task->copy_stack;
```

Comment:
`task->copy_stack` counts bytes from the high address (see `save_stack()`), so it should probably be something like:

```
*end = (char*)task->stkbuf + task->bufsz;
*start = *end - task->copy_stack;
```

Reply:
I don't follow. Looking at `save_stack`, I see this:
```C
...
    if (lastt->bufsz < nb) {
        buf = (void*)jl_gc_alloc_buf(ptls, nb);
        lastt->stkbuf = buf;
        lastt->bufsz = nb;
    }
    else {
        buf = lastt->stkbuf;
    }
    *pt = NULL; // clear the gc-root for the target task before copying the stack for saving
    lastt->copy_stack = nb;
    lastt->sticky = 1;
    memcpy_a16((uint64_t*)buf, (uint64_t*)frame_addr, nb);
...
```
So data starts at `lastt->stkbuf` and its length is `lastt->copy_stack`. What am I missing?

Reply:
Oh, I just realized that I confused the live stack with the saved stack. The copy is from `frame_addr`, but is put at the beginning of the buffer. Nevermind.